gcalbot/gcalbot/util.go

```diff
@@ -3,47 +3,132 @@ package gcalbot
+			return "", err
+		}
+		// set timezone
+		location, err := time.LoadLocation(timezoneID)
+		if err != nil {
+			return "", err
+		}
+		start = start.In(location)
+		end = end.In(location)
+	} else if startDateTime.Date != "" && endDateTime.Date != "" {
+		// this is an all day event
+		isAllDay = true
+		start, err = time.Parse("2006-01-02", startDateTime.Date)
+		if err != nil {
+			return "", err
+		}
+		end, err = time.Parse("2006-01-02", endDateTime.Date)
+		if err != nil {
+			return "", err
+		}
+		end = end.Add(-24 * time.Hour) // the google API sets the end day to the day after, so compensate by one day
+	} else {
+		return "", fmt.Errorf("invalid dates: start: %+v, end: %+v", startDateTime, endDateTime)
+	}
 
 	startYear, startMonth, startDay := start.Date()
 	endYear, endMonth, endDay := end.Date()
 
 	var startTime string
-	if start.Minute() == 0 {
-		startTime = start.Format("3pm")
-	} else {
-		startTime = start.Format("3:04pm")
-	}
-
 	var endTime string
-	if end.Minute() == 0 {
-		endTime = end.Format("3pm")
-	} else {
-		endTime = end.Format("3:04pm")
+	if !isAllDay {
+		if format24HourTime {
+			startTime = start.Format("15:04")
+			endTime = end.Format("15:04")
+		} else {
+			if start.Minute() == 0 {
+				startTime = start.Format("3pm")
+			} else {
+				startTime = start.Format("3:04pm")
+			}
+			if end.Minute() == 0 {
+				endTime = end.Format("3pm")
+			} else {
+				endTime = end.Format("3:04pm")
+			}
+		}
 	}
 
-	startTimezone, _ := start.Zone()
-	// endTimezone, _ := start.Zone()
-	// TODO(marcel): handle different timezones as a possibility
-
 	if startYear == endYear && startMonth == endMonth && startDay == endDay {
-		return fmt.Sprintf("%s %s - %s (%s)",
-			start.Format("Mon Jan 2, 2006"), startTime, endTime, startTimezone)
+		if isAllDay {
+			return start.Format("Mon Jan 2, 2006"), nil
+		} else {
+			return fmt.Sprintf("%s %s - %s (%s)",
+				start.Format("Mon Jan 2, 2006"), startTime, endTime, start.Format("MST")), nil
+		}
 	} else if startYear == endYear {
-		return fmt.Sprintf("%s %s - %s %s (%s)",
-			start.Format("Mon Jan 2"), startTime, end.Format("Mon Jan 2, 2006"), endTime,
-			startTimezone)
+		if isAllDay {
+			return fmt.Sprintf("%s - %s",
+				start.Format("Mon Jan 2"), end.Format("Mon Jan 2, 2006")), nil
+		} else {
+			return fmt.Sprintf("%s %s - %s %s (%s)",
+				start.Format("Mon Jan 2"), startTime, end.Format("Mon Jan 2, 2006"), endTime,
+				start.Format("MST")), nil
+		}
 	} else {
-		return fmt.Sprintf("%s %s - %s %s (%s)",
-			start.Format("Mon Jan 2, 2006"), startTime, end.Format("Mon Jan 2, 2006"), endTime,
-			startTimezone)
+		if isAllDay {
+			return fmt.Sprintf("%s - %s",
+				start.Format("Mon Jan 2, 2006"), end.Format("Mon Jan 2, 2006")), nil
+		} else {
+			return fmt.Sprintf("%s %s - %s %s (%s)",
+				start.Format("Mon Jan 2, 2006"), startTime, end.Format("Mon Jan 2, 2006"), endTime,
+				start.Format("MST")), nil
+		}
 	}
 }
 
+// TODO(marcel): implement backoff for api calls
```

Comment:
remove?

Reply:
Should I remove the commented out code entirely?

Reply:
yeah